molecule/driver/linode.py

```diff
@@ -0,0 +1,160 @@
+#  Copyright (c) 2015-2018 Cisco Systems, Inc.
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a copy
+#  of this software and associated documentation files (the "Software"), to
+#  deal in the Software without restriction, including without limitation the
+#  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+#  sell copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+#  DEALINGS IN THE SOFTWARE.
+
+from molecule import logger
+from molecule.driver import base
+
+from molecule import util
+
+LOG = logger.get_logger(__name__)
+
+
+class Linode(base.Base):
+    """
+    The class responsible for managing `Linode`_ instances.  `Linode`_
+    is `not` the default driver used in Molecule.
+
+    Molecule leverages Ansible's `linode_module`_, by mapping variables
+    from `molecule.yml` into `create.yml` and `destroy.yml`.
+
+    .. important::
+
+        Please note, the Ansible Linode module is currently using the deprecated
```

Comment:
Will swapping to v4 change how people *use* this driver, ie will there be any break to backwards compatibility?

Reply:
Yes, linode_v4 will probably break backwards compat. My plan was to deprecate the current linode module (since it uses deprecated API bindings on the Linode side) but that won't happen in the short-term since the new linode_v4 functionality won't catch up for a while. Assumed we'd roll out a linode_v4 driver alongside and then push a new major version to make the break when the existing linode module goes away.